utilities/blob_db/blob_db_impl.cc

```diff
@@ -1966,13 +1969,12 @@ bool BlobDBImpl::ShouldGCFile(std::shared_ptr<BlobFile> bfile, uint64_t now,
     return false;
   }
 
-  bool ret = bfile->BlobFileNumber() == last_id;
-  if (ret) {
-    *reason = "eligible last simple blob file";
-  } else {
-    *reason = "not eligible since not last simple blob file";
+  if (is_oldest_simple_blob_file) {
+    *reason = "out of space and is the oldest simple blob file";
```

Comment:
We probably can call them ttl blob file and non-ttl blob file. We store keys with TTL and without TTL in different blob files. They don't have difference in terms of file format. There is a subtype flag in the file header to tell whether the keys in the file has ttl or not.
https://github.com/facebook/rocksdb/blob/master/utilities/blob_db/blob_log_format.cc#L186